File: cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int cents;

    // Prompt the user for change owed, in cents
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);

    int quarters = cents / 25; // Calculate how many quarters you should give customer
    cents %= 25;               // Subtract the value of those quarters from cents

    int dimes = cents / 10; // Calculate how many dimes you should give customer
    cents %= 10;            // Subtract the value of those dimes from remaining cents

    int nickels = cents / 5; // Calculate how many nickels you should give customer
    cents %= 5;              // Subtract the value of those nickels from remaining cents

    int pennies = cents; // Calculate how many pennies you should give customer

    int total_coins = quarters + dimes + nickels +
                      pennies;   // Sum the number of quarters, dimes, nickels, and pennies used
    printf("%d\n", total_coins); // Print that sum

    return 0;
}

File: Code_Python/santimental-cash/cash.py
from cs50 import get_float


def main():
    # Continuously prompt the user for a valid positive amount of change
    while True:
        change_dollars = get_float("Change owed: ")
        if change_dollars >= 0:
            break

    # Convert dollars to cents
    change_cents = round(change_dollars * 100)

    # Initialize coin counter
    coins = 0

    # Define the coin values in cents
    quarters = 25
    dimes = 10
    nickels = 5
    pennies = 1

    # Calculate the minimum number of coins
    coins += change_cents // quarters
    change_cents %= quarters

    coins += change_cents // dimes
    change_cents %= dimes

    coins += change_cents // nickels
    change_cents %= nickels

    coins += change_cents // pennies
    change_cents %= pennies

    # Output the minimum number of coins
    print(coins)


if __name__ == "__main__":
    main()
